torch/nn/modules/module.py

```diff
@@ -935,8 +926,10 @@ def __getattr__(self, name: str) -> Union[Tensor, 'Module']:
             modules = self.__dict__['_modules']
             if name in modules:
                 return modules[name]
-        raise ModuleAttributeError("'{}' object has no attribute '{}'".format(
-            type(self).__name__, name))
+        # prevent raising AttributeError for function when local variables raise AttributeError
+        # and the __getattr__ is recalled. Instead, raise AttributeError for the local variable
+        # related Python issue: https://bugs.python.org/issue37379
+        super().__getattribute__(name)
```

Comment:
I think it would be easier to understand by implementing `__getattribute__` instead of `__getattr__`:

```python
def __getattribute__(self, name: str) -> Any:
    try:
        return super().__getattribute__(name)
    except AttributeError as e:
        # If the attribute doesn't exist, look for a parameter, buffer, or module
        # with the name.
        if '_parameters' in self.__dict__:
            _parameters = self.__dict__['_parameters']
            if name in _parameters:
                return _parameters[name]
        if '_buffers' in self.__dict__:
            _buffers = self.__dict__['_buffers']
            if name in _buffers:
                return _buffers[name]
        if '_modules' in self.__dict__:
            modules = self.__dict__['_modules']
            if name in modules:
                return modules[name]

        # Note: Purposely reuse the specific error thrown from the original attempt to
        # access the property. This ensures that the error message is correct for the
        # case where a property method attempts to call a non-existent property, for example.
        raise e
```

This fixes the problem described in the original issue, and doesn't make extra calls and rely on some (possibly buggy) internal python implementation details.

(see https://stackoverflow.com/questions/25115366/property-and-getattr-compatibility-issue-with-attributeerror for discussion & the proposed solution)

Reply:
Yeah. This is definitely easier to understand. I did see this suggestion on stackoverflow when I was working on this. I think this was also suggested on the previous similar issue.
However, I thought since the current implementation was already based on the silent AttributeError from getattribute, I thought just adding a line would be the easiest fix.
I can change it if you want me to.

Reply:
I tried using the code you suggested and seems like we'd have to make a lot more changes to make it work without `__getattr()__` function. I guess, we have to to at least have the function.

Reply:
Ah interesting- out of curiosity, what are some of the other changes that would be required? I'm reconsidering whether I think changing to `__getattribute__` is the way to go and want to understand any BC implications.

Reply:
There are a few files:
`test_fx.py`, `test_numeric_suite_fx.py`,  `test_quantize_fx.py`  (i think there's a few more)
where the changes would have to be made. It looks like all the error are similar and come from the following line:
`orig_getattr = torch.nn.Module.__getattr__`
I'll try to check if simply changing these to `torch.nn.Module.__getattribute__` will work.

Reply:
seems like changing to `torch.nn.Module.__getattribute__` or just explicitly having
`
def __getattr__(self):
    pass
` 
in the Module class both will work for those errors.